llvm/include/llvm/Support/GenericDomTree.h

```diff
@@ -885,6 +874,12 @@ class DominatorTreeBase {
         NewBBIDom = findNearestCommonDominator(NewBBIDom, PredBlocks[i]);
     }
 
+    bool NewBBDominatesNewBBSucc =
+      llvm::none_of(inverse_children<N>(NewBBSucc), [=](const auto *Pred) {
```

Comment:
For locally used lambdas, I'd usually suggest `[&]` (applying to here and elsewhere in thepatch) to reduce the risk of referencing something expensive and copying by accident?

Reply:
Good point here. It is also implicitly capturing `this` which is deprecated in C++20.